spec/unit/daemon_spec.rb

```diff
@@ -79,6 +79,17 @@ def run_loop(jobs)
       daemon.start
       expect(scheduler.jobs[0]).not_to be_enabled
     end
+
+    it "recalculates splay if splaylimit changes" do
+      # Set file timeout so the daemon reparses
+      Puppet[:filetimeout] = 1
+      Puppet[:splay] = true
+      allow(agent).to receive(:run)
+      daemon.start
+      first_splay = scheduler.jobs[1].splay
+      scheduler.jobs[0].run(Time.now)
```

Comment:
When the reparse job runs it recalculates splay using `rand(limit + 1)` where `limit` defaults to the `Puppet[:runinterval]`. So there's a 1 in 30*60 chance `rand` will return the same value as before causing the test to fail. Suggest stubbing the `rand` call before calling `run` so the test is deterministic:

```suggestion
      allow(Kernel).to receive(:rand).and_return(<some number that isn't first_splay>)
```